riscv/src/compiler.rs

```diff
@@ -1128,10 +1135,15 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             vec![format!("jump_and_link_dyn {rs};")]
         }
         "call" => {
-            if let [label] = args {
-                vec![format!("call {};", argument_to_escaped_symbol(label))]
-            } else {
-                panic!()
+            // Depending on what symbol is called, the call is replaced by a
+            // powdr asm call, or a call to a coprocessor if a special function
+            // has been recognized.
+            match args {
+                [Argument::Expression(Expression::Symbol(label))] if label == "poseidon_hash" => {
```

Comment:
I think it would be better to make this configurable - the compiler should not need to know about which such functions exist and how they are implemented.

Reply:
that makes sense, and I completely agree. I have just pushed another commit that does that, but it requires the experimental `if_let_guard` and it's still located in the compiler crate at the moment.

Reply:
which breaks the tests in CI :(